Feat: ECS compatible default behavior (#89)
logstash-plugins/logstash-filter-translate: docs/index.asciidoc

copy/edit/rename or create/rename mechanism to avoid the refresh code from
processing half-baked dictionary content.

> ==== Compatibility with the Elastic Common Schema (ECS)

Reviewer: ```suggestion
[id="plugins-{type}s-{plugin}-ecs_metadata"]
==== Compatibility with the Elastic Common Schema (ECS)
```

Reviewer: This is the anchor we've used on other ECS plugins.  If the word "metadata" is too off target for this application, we can pick another word as long as it doesn't create a duplicate anchor.

> The plugin acts as an in place translator if `field` and `target` are the same

Reviewer: ```suggestion
The plugin acts as an in-place translator if `field` and `target` are the same
```

> and does not produce any new event fields.

Reviewer: >The plugin acts as an in-place translator if `field` and `target` are the same and does not produce any new event fields.

This statement takes a minute to parse and could be clearer with a slight rewording:

I think this is what you mean: 
If `field` and `target` are the same, the plugin acts as an in-place translator  and does not produce any new event fields.

===== `destination`

> 

Reviewer: ```suggestion
deprecated[3.3.0]
```

Reviewer: This is how we format deprecation notices. Looks like this: 
<img width="860" alt="Screen Shot 2021-06-22 at 3 13 16 PM" src="https://user-images.githubusercontent.com/35154725/122985656-70f58780-d36c-11eb-829d-6acbfec1ac89.png">

Here's another option. 
`deprecated[3.3.0, The default value has been changed to `false`. In 8.0.0 this setting will be removed]`
It gives you a place to put more info, but it doesn't look as good (because it renders bigger than the heading). Looks like this: https://www.elastic.co/guide/en/logstash/current/plugins-inputs-beats.html#plugins-inputs-beats-add_hostname.

Author: Thanks, I've added the *deprecated* marker. 
There's also a new `source` option (as a replacement for the now deprecated `field`), let me know if I should tune the deprecation warning further ... :cake:

Reviewer: Your deprecation notices look like this: <img width="621" alt="Screen Shot 2021-06-24 at 4 45 53 PM" src="https://user-images.githubusercontent.com/35154725/123330223-c0be8500-d50b-11eb-888c-826f5308501e.png">

Adding a link to the option they should use instead is a helpful. ❤️

===== `override`

  * Value type is <<boolean,boolean>>
  * Default value depends on whether in-place translation is being used

> If the destination (or target) field already exists, this configuration item specifies

Reviewer: ```suggestion
If the destination (or target) field already exists, this configuration option controls
```

> whether the filter should skip translation (default behavior) or overwrite the target
> field value with the new translation value.

Reviewer: ```suggestion
whether the filter skips translation (default behavior) or overwrites the target
field value with the new translation value.
```

will remain after the merge; `replace` causes the whole dictionary to be replaced
with a new one (deleting all entries of the old one on update).

[id="plugins-{type}s-{plugin}-target"]
===== `target`

  * Value type is <<string,string>>
  * Default value depends on whether <<plugins-{type}s-{plugin}-ecs_compatibility>> is enabled:
    ** ECS Compatibility disabled: `"translation"`
    ** ECS Compatibility enabled: defaults to the same value as `field`

The target field you wish to populate with the translated code.

> Set this to the same value as source `field` causes the plugin to do a substitution,
> in this case filter will always succeed. This will clobber the old value of the source field!

Reviewer: ```suggestion
If you set this value to the same value as source `field`, the plugin does a substitution, and
the filter will succeed. This will clobber the old value of the source field!
```

Reviewer: Is that sometimes/always/never preferred behavior? 

If this might be the behavior somebody wants, maybe something like: "The filter will succeed, and the old value of the source field will be replaced. "

Author: Thanks, should be fine as you suggested ... mentioning "always succeeds" was unnecessary confusing to start with
